test/ziggurat/messaging/producer_test.clj

```diff
@@ -151,15 +139,26 @@
             expected-dead-set-message (assoc message-payload :retry-count (retry-count-config))]
         (producer/retry retry-message-payload)
         (let [message-from-mq (rmq/get-msg-from-dead-queue "default")]
-          (is (= expected-dead-set-message message-from-mq))))))
+          (is (= (bytes-to-str expected-dead-set-message) (bytes-to-str message-from-mq)))))))
 
-  (testing "message with no retry count will publish to delay queue"
+  (testing "message with no retry count will publish to dead queue"
```

Comment:
Should we have a test  for `0` retry-count as well?

Since, that's also part of the logic.

Reply:
It's already there